circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -251,12 +251,14 @@ void update_end_values(DummyComposer& composer,
     // }
 }
 
-void validate_this_private_call_hash(DummyComposer& composer, PrivateInputs<NT> const& private_inputs)
+void validate_this_private_call_hash(DummyComposer& composer,
+                                     PrivateInputs<NT> const& private_inputs,
+                                     KernelCircuitPublicInputs<NT>& public_inputs)
 {
-    const auto& start = private_inputs.previous_kernel.public_inputs.end;
+
     // TODO: this logic might need to change to accommodate the weird edge 3 initial txs (the 'main' tx, the 'fee' tx,
     // and the 'gas rebate' tx).
-    const auto popped_private_call_hash = array_pop(start.private_call_stack);
+    const auto popped_private_call_hash = array_pop(public_inputs.end.private_call_stack);
```

Comment:
Okay so this switches it to pop from the accumulated/end callstack instead of the one in the private inputs. The private kernel uses the one from private inputs. Is there a difference?